### PR TITLE
Add cleanup function and tests for px.datepicker

### PR DESCRIPTION
**Pull request by Akswii**
Added code to clean up to delete old flatpickr instances when datepicker.init() is called more than once.

### CODE REVIEW

File: src/px-script/main/datepicker/index.js [hunk 1]
```diff
@@ -57,6 +57,10 @@ const _createDatepicker = datepicker => {
     return datepickerObj;
 };
 
+const _destroyDatepickers = () => {
+    _datepickers.forEach(datepicker => datepicker.destroy());
+};
+
 const init = id => {
     if (id) {
```
Comment (erikhallaraaker) [inline src/px-script/main/datepicker/index.js]: Should previous datepickers be removed regardless when `init` is invoked ? If so move this repeated statement to the top and remove statements below. If not ignore this and add tests for it.

```suggestion
    if (_datepickers.length) { _destroyDatepickers(); }
    if (id) {
```

Comment (Akswii) [reply src/px-script/main/datepicker/index.js]: If i move the test to the top of init then old datepickers would be destroyed if you sent a specific ID. Which raises another relevant question; should datepickers be destroyed when you init a single specific ID? Should probably only be done when you init ALL datepickers.

Comment (Akswii) [reply src/px-script/main/datepicker/index.js]: On second thought datepickers would not be destroyed on invalid ID due to the test right before. But datepickers should not be destroyed on ID-init anyways.

Comment (erikhallaraaker) [reply src/px-script/main/datepicker/index.js]: Right now you are destroying all datepickers if invoking `init` with a valid ID. If that should not be the case, you should remove the `_destroyDatepickers();` line from the `if (id)` block, and add tests to make sure they are not destroyed :)

Comment (Akswii) [reply src/px-script/main/datepicker/index.js]: Comment to my self: Destroy the datepicker matching the passed ID.

File: src/px-script/main/datepicker/index.js [hunk 2]
```diff
@@ -67,6 +72,11 @@ const init = id => {
             return null;
         }
 
+        _datepickers.forEach((d, i) => (d.element.id === id
+            ? (d.destroy(), _datepickers.splice(i, 1))
```
Comment (erikhallaraaker) [inline src/px-script/main/datepicker/index.js]: Save that comma yo

```suggestion
            ? _datepickers.splice(i, 1)[0].destroy()
```

File: src/px-script/main/datepicker/index.js [hunk 3]
```diff
@@ -57,6 +57,11 @@ const _createDatepicker = datepicker => {
     return datepickerObj;
 };
 
+const _destroyDatepickers = () => {
+    _datepickers.forEach(datepicker => datepicker.destroy());
```
Comment (erikhallaraaker) [inline src/px-script/main/datepicker/index.js]: With this you can remove the next line
```suggestion
    _datepickers = _datepickers.filter(datepicker => datepicker.destroy());
```

**Pull request merged**